iina/Preference.swift

```diff
@@ -959,4 +955,9 @@ struct Preference {
     return T.init(key: key) ?? T.defaultValue
   }
 
+  static func keyHasBeenPersisted(_ key: Key) -> Bool {
+    let identifier = InfoDictionary.shared.bundleIdentifier
+    guard let persisted = ud.persistentDomain(forName: identifier) else { return false }
```

Comment:
I suggest make it to be
``` swift
static let persisted = ud.persistentDomain(forName: identifier)
guard let _ = persisted else { return false }
```
to prevent retrieving the whole UserDefault multiple times.

Reply:
If so, `persisted` will only be updated when IINA starts. If someone find this function in the future and try to use it, they may encounter problems if they updated the preference when IINA is running. I actually suggest moving this part to `LegacyMigration` where it's being used, as an inner function or block.